samples/csharp/MicrosoftEduGraphSamples/Workflows/AssignmentWorkflow.cs

```diff
@@ -64,5 +67,95 @@ public async Task<IEnumerable<EducationAssignment>> GetMeAssignmentsFromNonArchi
+                Console.WriteLine($"Assignment created successfully {assignmentDraft.Id} in state {assignmentDraft.Status}");
+
+                // Teacher publishes the assignment to make it appears in the student's list
+                assignmentInactive = await MicrosoftGraphSDK.Assignment.PublishAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine($"Assignment {assignmentInactive.Id} publish in process");
+
+                // Verify assignment state, publish is completed until state equals "Assigned"
+                while (assignmentInactive.Status != EducationAssignmentStatus.Assigned && retries <= MAX_RETRIES)
+                {
+                    // Print . in the log to show that the call is being retried
+                    Console.WriteLine(".");
+
+                    assignmentInactive = await MicrosoftGraphSDK.Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+
+                    // If you are calling this code pattern in Backend agent of your service, then you want to retry the work after some time. The sleep here is just an example to emulate the delay
+                    Thread.Sleep(2000);
+                    retries++;
+                }
+                Console.WriteLine($"Assignment {assignmentInactive.Id} publish is completed. Status: {assignmentInactive.Status}");
+
+                // Deactivate the Assignment
+                assignmentInactive = await MicrosoftGraphSDK.Assignment.DeactivateAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine($"Assignment {assignmentInactive.Id} Deactivated");
+
+                // Publishing an Assignment
+                assignmentId = assignmentAssigned.Id;
+                assignmentAssigned = await MicrosoftGraphSDK.Assignment.PublishAsync(graphClient, _config["classId"], assignmentId);
+                Console.WriteLine($"Assignment {assignmentAssigned.Id} publish in process");
+
+                // Verify assignment state, publish is completed until state equals "Assigned"
+                while (assignmentAssigned.Status != EducationAssignmentStatus.Assigned && retries <= MAX_RETRIES)
+                {
+                    // Print . in the log to show that the call is being retried
+                    Console.WriteLine(".");
+
+                    assignmentAssigned = await MicrosoftGraphSDK.Assignment.GetAssignmentAsync(graphClient, _config["classId"], assignmentId);
+
+                    // If you are calling this code pattern in Backend agent of your service, then you want to retry the work after some time. The sleep here is just an example to emulate the delay
+                    Thread.Sleep(2000);
+                    retries++;
+                }
+                Console.WriteLine($"Assignment {assignmentAssigned.Id} publish is completed. Status: {assignmentAssigned.Status}");
```

Comment:
remove this code and just call the new global method you will create to check the PUBLISH process (from my comment above).

Do the same for any other sample that is publishing assignments.

Reply:
Updated the method in Global